backend/routes/graph.py

```diff
@@ -29,166 +35,308 @@ def subgraph():
     return list(REL_GRAPH.edges)
 
 
-@router.post("/subgraph")
-async def subgraph_post(request: Request, id: Union[List[int], None] = None) -> List:
+@router.get("/indented-concept-list")
+async def indented_concept_list(request: Request, id: List[int] = Query(...)):   # id is a list of concept ids
+    return await indented_concept_list_post(request=request, id=id)
+
+
+@router.post("/indented-concept-list")
+async def indented_concept_list_post(request: Request, id: Union[List[int], None] = None) -> List:
     rpt = Api_logger()
     await rpt.start_rpt(request, params={'concept_ids': id})
 
     try:
-        sg = connected_subgraph_from_nodes(id, REL_GRAPH, REL_GRAPH_UNDIRECTED)
-        edges = [(str(e[0]), str(e[1])) for e in sg.edges]
-        await rpt.finish(rows=len(edges))
+        # new way to do this
+        paths = all_paths(REL_GRAPH, id)
+        tree = paths_as_indented_tree(paths)
+        await rpt.finish(rows=len(tree))
     except Exception as e:
         await rpt.log_error(e)
         raise e
-    return edges
+    return tree
 
 
-@router.get("/subgraph")
-async def subgraph(request: Request, id: List[int] = Query(...)):   # id is a list of concept ids
-    return await subgraph_post(request=request, id=id)
-
-
-@router.get("/hierarchy")
-def hierarchy(id: List[int] = Query(...)):   # id is a list of concept ids
+def all_paths(g: networkx.DiGraph, nodes: List[int]) -> List[List[int]]:
```

Comment:
`all_paths()`: I get what this function does and I heard your explanation earlier. It's still not 100% in my mind how it is all glued together and what the end result looks like but I'm sure that will become clear to me in time. When this is done, I'll want to see the end result.

- [ ] @joeflack4: Gain full understanding of this and then mark resolved.

Reply:
There's inline documentation (generated by chatgpt, but pretty good). But to give a clearer idea of the mechanics, or at least the data shapes, I created [a new documentation file](https://github.com/jhu-bids/TermHub/blob/fixing-concept-hierarchy/docs/graph.md)